Detectors/TPC/qc/src/PID.cxx

```diff
@@ -8,50 +8,108 @@
+  mHist1D.emplace_back("hdEdxMax", "; dEdxMax (a.u.); counts", 200, 0, 200);   //| mHist1D[2]
+  mHist1D.emplace_back("hPhi", "; #phi (rad); counts", 180, -M_PI, M_PI);      //| mHist1D[3]
+  mHist1D.emplace_back("hTgl", "; tan#lambda; counts", 60, -2, 2);             //| mHist1D[4]
+  mHist1D.emplace_back("hSnp", "; sin p; counts", 60, -2, 2);                  //| mHist1D[5]
+
+  mHist2D.emplace_back("hdEdxVsPhi", "dEdx (a.u.) vs #phi (rad); #phi (rad); dEdx (a.u.)", 180, -M_PI, M_PI, 300, 0, 300); //| mHist2D[0]
+  mHist2D.emplace_back("hdEdxVsTgl", "dEdx (a.u.) vs tan#lambda; tan#lambda; dEdx (a.u.)", 60, -2, 2, 300, 0, 300);        //| mHist2D[1]
+  mHist2D.emplace_back("hdEdxVsncls", "dEdx (a.u.) vs ncls; ncls; dEdx (a.u.)", 80, 0, 160, 300, 0, 300);                  //| mHist2D[2]
+
+  const auto logPtBinning = helpers::makeLogBinning(30, 0.1, 10);
+  if (logPtBinning.size() > 0) {
+    mHist2D.emplace_back("hdEdxVsp", "dEdx (a.u.) vs p (G#it{e}V/#it{c}); p (G#it{e}V/#it{c}); dEdx (a.u.)", logPtBinning.size() - 1, logPtBinning.data(), 300, 0, 300); //| mHist2D[3]
+  }
+  //mHist2D.emplace_back("hdedxVsphiMIPA","; #phi (rad); dedx (a.u.)", 180,-M_PI,M_PI,25,35,60);  //| mHist2D[4]
+  //mHist2D.emplace_back("hdedxVsphiMIPC","; #phi (rad); dedx (a.u.)", 180,-M_PI,M_PI,25,35,60);  //| mHist2D[5]
 }
 
+//______________________________________________________________________________
 void PID::resetHistograms()
 {
   for (auto& hist : mHist1D) {
     hist.Reset();
   }
+  for (auto& hist : mHist2D) {
+    hist.Reset();
+  }
 }
 
+//______________________________________________________________________________
 bool PID::processTrack(o2::tpc::TrackTPC const& track)
 {
   // ===| variables required for cutting and filling |===
   const auto p = track.getP();
-  const auto dEdx = track.getdEdx().dEdxTotTPC;
-
-  // ===| cuts |===
-  // hard coded cuts. Should be more configural in future
-  if (std::abs(p - 0.5) > 0.05) {
-    return false;
-  }
-
-  if (dEdx > 70) {
-    return false;
-  }
+  const auto dEdxTot = track.getdEdx().dEdxTotTPC;
+  const auto dEdxMax = track.getdEdx().dEdxMaxTPC;
+  const auto phi = track.getPhi();
+  const auto tgl = track.getTgl();
+  const auto snp = track.getSnp();
+  const auto nclusters = track.getNClusterReferences();
 
   // ===| histogram filling |===
-  mHist1D[0].Fill(dEdx);
+  mHist1D[0].Fill(nclusters);
+  mHist1D[1].Fill(dEdxTot);
+  mHist1D[2].Fill(dEdxMax);
+  mHist1D[3].Fill(phi);
+  mHist1D[4].Fill(tgl);
+  mHist1D[5].Fill(snp);
+
+  mHist2D[0].Fill(phi, dEdxTot);
+  mHist2D[1].Fill(tgl, dEdxTot);
+  mHist2D[2].Fill(nclusters, dEdxTot);
+  mHist2D[3].Fill(p, dEdxTot);
 
   return true;
 }
+
+//______________________________________________________________________________
+void PID::setNiceStyle()
+{
+  gStyle->SetPalette(kCividis);
+
+  for (auto& hist : mHist2D) {
+    helpers::setStyleHistogram2D(hist);
+  }
```

Comment:
As written above, perhaps the helpers can have a function which takes the vector.

Reply:
see above